File: bindings/elixir/lib/proven/safe_currency.ex
# SPDX-License-Identifier: PMPL-1.0-or-later
# Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <jonathan.jewell@open.ac.uk>

defmodule Proven.SafeCurrency do
  @moduledoc """
  Safe currency operations via libproven FFI.

  Currency parsing and formatting are performed by the Idris 2 verified
  core via `proven_currency_parse` and `proven_currency_format`.

  Note: Full currency operations via libproven require marshaling the
  `CurrencyResult` struct through the NIF. These will be added as the
  NIF layer expands.
  """

  # Note: proven_currency_parse and proven_currency_format involve
  # complex CurrencyResult struct marshaling. These will be added
  # when the NIF layer supports the currency C structs.
end

File: bindings/deno/src/safe_currency.ts
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <jonathan.jewell@open.ac.uk>

/**
 * SafeCurrency - Currency operations.
 *
 * Thin FFI wrapper: all computation delegates to libproven (Idris 2 + Zig).
 * Note: proven_currency_parse returns CurrencyResult and
 * proven_currency_format takes [3]u8 + u8 structs, both requiring
 * buffer-based marshaling. This module provides the FFI infrastructure.
 *
 * @module
 */

// Note: proven_currency_* functions require complex struct marshaling.
// This module is a placeholder that will be fully wired when the
// buffer protocol for struct-by-value parameters is available.
// Currently no reimplemented logic.

export { getLib, isAvailable } from './ffi.ts';
